cores/arduino/Print.h

```diff
@@ -54,6 +57,10 @@ class Print
       return write((const uint8_t *)buffer, size);
     }
 
+    // default to zero, meaning "a single write may block"
+    // should be overriden by subclasses with buffering
+    virtual int availableForWrite() { return 0; }
```

Comment:
Based on testing, adding the virtual doesn't seem to make a difference, we just needed to change the `size_t` to `int` in I2S.